vendor/bat-native-ads/src/bat/ads/internal/federated/log_entries/impression_served_at_covariate_log_entry.h

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_FEDERATED_LOG_ENTRIES_IMPRESSION_SERVED_AT_COVARIATE_LOG_ENTRY_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_FEDERATED_LOG_ENTRIES_IMPRESSION_SERVED_AT_COVARIATE_LOG_ENTRY_H_
+
+#include <string>
+
+#include "bat/ads/internal/federated/covariate_log_entry.h"
+
+namespace base {
+class Time;
+}  // namespace base
+
+namespace ads {
+
+class ImpressionServedAtCovariateLogEntry final : public CovariateLogEntry {
+ public:
+  ImpressionServedAtCovariateLogEntry();
+  ImpressionServedAtCovariateLogEntry(
+      const ImpressionServedAtCovariateLogEntry&) = delete;
+  ImpressionServedAtCovariateLogEntry& operator=(
+      const ImpressionServedAtCovariateLogEntry&) = delete;
+  ~ImpressionServedAtCovariateLogEntry() override;
+
+  void SetLastImpressionAt(const base::Time& time);
+
+  // CovariateLogEntry
+  mojom::DataType GetDataType() const override;
+  mojom::CovariateType GetCovariateType() const override;
+  std::string GetValue() const override;
+
+ private:
+  base::Time impression_served_at_;
```

Comment:
we couldn't use forward declaration when declare class member

Reply:
Changed it based on @tmancey feedback. What are your thoughts @tmancey, Should I change it back?

Reply:
just to double check forward decl does compile? Should I still change?

Reply:
I think we should add `#include "base/time/time.h"` here.
https://stackoverflow.com/questions/553682/when-can-i-use-a-forward-declaration

Reply:
fixed